src/util.ts

```diff
@@ -38,12 +39,77 @@ export function nop() {
 /**
  * Find the tsconfig.json, read it, and return parsed contents.
  * @param rootDir Directory where the tsconfig.json should be found.
+ * If the tsconfig.json file has an "extends" field hop down the dependency tree until it ends
  */
+
+
+export interface ConfigFile {
+  files?: string[];
+  compilerOptions?: {};
+  include?: string[];
+  exclude?: string[];
+}
+
 export async function getTSConfig(
-    rootDir: string, customReadFilep?: ReadFileP): Promise<{}> {
+    rootDir: string, customReadFilep?: ReadFileP): Promise<ConfigFile> {
+  // array to hold accessed files, used to check for circular references
+
+
+
   const tsconfigPath = path.join(rootDir, 'tsconfig.json');
+
   customReadFilep = customReadFilep || readFilep;
   const json = await customReadFilep(tsconfigPath, 'utf8');
-  const contents = JSON.parse(json);
-  return contents;
+  let contents = JSON.parse(json);
+
+  const readArr = ['tsconfig.json'];
+  if (contents['extends']) {
+    const extension =
+        await getExtension(contents['extends'], customReadFilep, readArr);
+    contents = combineTSConfig(extension, contents);
+  }
+  return Promise.resolve(contents);
 }
+
+async function getExtension(
+    filePath: string, customReadFilep: ReadFileP,
+    readFiles: string[]): Promise<ConfigFile> {
+  customReadFilep = customReadFilep || readFilep;
+  if (!filePath) {
+    throw new Error('Undefined passed in');
+  }
+  if (readFiles.indexOf(filePath) !== -1) {
+    // did I throw the error correctly?
+    throw new Error('Circular Reference Detected');
+  }
+  readFiles.push(filePath);
+  const json = await customReadFilep(filePath, 'utf8');
+  let contents = JSON.parse(json);
+
+  if (contents['extends']) {
+    const nextFile =
+        await getExtension(contents['extends'], customReadFilep, readFiles);
+    contents = combineTSConfig(nextFile, contents);
+  }
+  // console.log(contents);
+  return Promise.resolve(contents);
+}
+
+// the inherited config file overwrites the base config file's "files",
+// "include", and "exclude" fields and combines compiler options
+function combineTSConfig(base: ConfigFile, inherited: ConfigFile): ConfigFile {
+  // const TSProperties = ["compilerOptions", "files", "include", "exclude"];
+  const result = {
```

Comment:
I think we can just initialize to `{ compilerOptions: {} }`. If neither `base` nor `inherited` has any of the other fields defined, then an empty `files` (for example) might be semantically different than an undefined `files`.